app/models/authentication.rb

```diff
@@ -34,6 +34,41 @@ def self.new(*args, &block)
 
   serialize :options
 
+  include OwnershipMixin
+  include TenancyMixin
+
+  include ReservedMixin
+  reserve_attribute :evm_owner_id, :big_integer
+  reserve_attribute :miq_group_id, :big_integer
+  reserve_attribute :tenant_id,    :big_integer
+
+  virtual_belongs_to :evm_owner, :class_name => ::User
+  def evm_owner
+    User.find_by(:id => evm_owner_id)
+  end
+
+  def evm_owner=(owner)
+    update_attributes!(:evm_owner_id => owner.id) if owner
```

Comment:
For these, I think you want something like:

```ruby
def evm_owner=(owner)
  evm_owner_id = owner.id
end
```

The `reserved_rec` relation has `:autosave => true` which should save when the authentication is saved.  Otherwise we could end up with someone calling `authentication.evm_owner = new_owner`, not saving the record and the reserved record being updated anyway, right?

Reply:
Mm yeah good point